core/include/prometheus/detail/hash.h

```diff
@@ -0,0 +1,46 @@
+#pragma
```

Comment:
There's no need for that file to be a public header. Please move it to `core/src/detail/`.

Reply:
I could not understand that why we move `hash.h` file into `core/src/detail` folder.

If we compile this project to a static/shared library,  and provide `core/include` for users, how can users include `hash.h` file?

Reply:
Right now users don't have to see that file so we could hide it as an implementation detail and won't have to take care of API / ABI compatibility.